lib/src/airship_flutter.dart

```diff
@@ -295,11 +296,31 @@ class Airship {
     return _eventStreams[eventType];
   }
 
-  /// Initializes Airship with an [appKey] and [appSecret].
+  /// Initializes Airship with an [AirshipConfig] or
+  /// [appKey] and [appSecret].
   ///
   /// Returns true if Airship has been initialized, otherwise returns false.
-  static Future<bool> takeOff(String appKey, String appSecret) async {
-    Map<String, String> args = {"app_key": appKey, "app_secret": appSecret};
+  /// For backwards compatibility [keyConfig] can be appKey[String] or [AirshipConfig]
+  /// in next major release airship will migrate to [AirshipConfig]
+  static Future<bool> takeOff(final Object keyConfig,
+      [final String appSecret = ""]) async {
+    assert(
+      keyConfig is String || keyConfig is AirshipConfig,
+      "For backwards compatibility [appKey] can be [String] or [AirshipConfig]",
```

Comment:
This could be `Airship failed to takeOff! An appKey String or valid AirshipConfig are required.`?